noir-projects/noir-protocol-circuits/crates/types/src/public_keys.nr

```diff
@@ -96,13 +98,45 @@ impl Serialize<POINT_LENGTH> for TpkM {
     }
 }
 
-impl Empty for PublicKeys {
-    fn empty() -> Self {
+impl Default for PublicKeys {
+    // This information is duplicated in circuits.js/src/types/public_keys.ts
+    // We use this because empty will produce a point not on the curve.
+    // This is:
+    // "az_null_npk"
+    // "az_null_ivpk"
+    // "az_null_ovpk"
+    // "az_null_tpk"
+    // as bytes, hashed to curve using grumpkin::g1::affine_element::hash_to_curve(<X>, 0);
+    fn default() -> Self {
         PublicKeys {
-            npk_m: NpkM { inner: Point::empty() },
-            ivpk_m: IvpkM { inner: Point::empty() },
-            ovpk_m: OvpkM { inner: Point::empty() },
-            tpk_m: TpkM { inner: Point::empty() }
+            npk_m: NpkM {
```

Comment:
Should we store these values inside `noir-projects/noir-protocol-circuits/crates/types/src/constants.nr`. We store other magic values there and it will be easier to propogate across TS and CPP

They may need to be stored in coordinate form like
`global DEFAULT_NPK_M_X = 0x01498945581e0eb9f8427ad6021184c700ef091d570892c437d12c7d90364bbd"`